### ISSUE TITLE
no-shadow: Duplicated error messages

### ISSUE THREAD

**Issue by roadhump** [eslint/eslint#2706]
Now it seems that `no-shadow` rule outputs several identical error messages if variable was shadowed several times:

```js
var a = function(a) {
    return function(a) {
        return function() {
            var a = 123;
            return a;
        };
    };
};

```

```
  4:17  error  a is already declared in the upper scope  no-shadow
  5:20  error  a is already declared in the upper scope  no-shadow
  5:20  error  a is already declared in the upper scope  no-shadow
  7:16  error  a is already declared in the upper scope  no-shadow
  7:16  error  a is already declared in the upper scope  no-shadow
  7:16  error  a is already declared in the upper scope  no-shadow
```

I suppose there should be only three non-duplicated messages

**Comment by michaelficarra** [comment]
You're suggesting we only report the shadowing of the nearest declaration?

**Comment by roadhump** [comment]
I suggest that errors should be 
```
  4:17  error  a is already declared in the upper scope  no-shadow
  5:20  error  a is already declared in the upper scope  no-shadow
  7:16  error  a is already declared in the upper scope  no-shadow
```

As I see in code, rule does not provide any info about which upper declaration was shadowed, only which variable shadows it. I suggest to output only first case when variable shadows any upper declaration, because messages are identical if there are several cases.

**Comment by michaelficarra** [comment]
Yeah, that sounds better. :+1:.

**Comment by ilyavolodin** [comment]
Yes, I think reporting only the closest shadowing is a good idea.

**Comment by gcochard** [comment]
FWIW I think reporting all instances of a shadowed var is a good thing. I know all areas to fix rather than fixing one, only to find another, and another, etc.

Should this be an option for the rule? Something like `suppressDuplicates`

**Comment by ilyavolodin** [comment]
All of them are going to be reported, just not for each instance. If you look at the example above and expected result, you can see that all of the instances of shadowing is reported, but current version reports each shadowing issue for each individual error (so geometric progression).

**Comment by gcochard** [comment]
Ah, right. I missed that this was the case :+1:.
